### ISSUE TITLE
Throw a descriptive error when a custom validator doesn't return a promise

### ISSUE THREAD

**Issue by spruce-bruce** [synapsestudios/hapi-async-validation#5]
Right now I think things just fail weirdly and without explanation if you return anything that isn't a promise. We should throw an error that has more explanation like "Custom validators must always return promises", etc

**Comment by areida** [comment]
I don't think this is a concern anymore with `async/await` since an async function can return a non-promise and still be treated as a resolved promise, right?
